packages/components/data-table/src/cell.js

```diff
@@ -114,14 +113,14 @@ DataCell.propTypes = {
   isCondensed: PropTypes.bool,
   isTruncated: PropTypes.bool,
   shouldIgnoreRowClick: PropTypes.bool,
-  handleRowCollapseClick: requiredIf(
-    PropTypes.func,
-    (props) => props.isTruncated
-  ),
   shouldRenderCollapseButton: requiredIf(
```

Comment:
I wouldn't make this one required. It's only `true` when a DataCell is in the last column (and only in combination with `isTruncated`), but otherwise I think it can be omitted safely.